cli/config/app.babel.config.js

```diff
@@ -24,8 +24,22 @@ module.exports = {
 
         // Automatically include a React import when JSX is present
         require('babel-plugin-react-require'),
-
-        // Always build in "production" mode even when styled-jsx runtime may select "development"
-        [require('styled-jsx/babel'), { optimizeForSpeed: true }],
     ],
+    env: {
+        production: {
+            plugins: [
+                [require('styled-jsx/babel'), { optimizeForSpeed: true }],
+            ]
+        },
+        development: {
+            plugins: [
+                [require('styled-jsx/babel'), { optimizeForSpeed: true }],
```

Comment:
@varl should we turn `optimizeForSpeed` off here?

Reply:
Note this is for both apps and libraries, but I think it should be safe to build without `optimizeForSpeed` when the build output is in development...?

Reply:
Anyway, will merge this and we can consider later.

Reply:
The tricky thing is when production-mode components and development-mode components interact on the same page. Since styled-jsx is a singleton, that causes problems as production/development components inject CSS differently and are not compatible.